mat/basictypes_test.go

```diff
@@ -0,0 +1,114 @@
+// Copyright ©2015 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mat
+
+import "gonum.org/v1/gonum/blas/blas64"
+
+func asBasicMatrix(d *Dense) *basicMatrix              { return (*basicMatrix)(d) }
+func asBasicVector(d *VecDense) *basicVector           { return (*basicVector)(d) }
+func asBasicSymmetric(s *SymDense) *basicSymmetric     { return (*basicSymmetric)(s) }
+func asBasicTriangular(t *TriDense) *basicTriangular   { return (*basicTriangular)(t) }
+func asBasicBanded(b *BandDense) *basicBanded          { return (*basicBanded)(b) }
+func asBasicSymBanded(s *SymBandDense) *basicSymBanded { return (*basicSymBanded)(s) }
+func asBasicTriBanded(t *TriBandDense) *basicTriBanded { return (*basicTriBanded)(t) }
+func asBasicDiagonal(d *DiagDense) *basicDiagonal      { return (*basicDiagonal)(d) }
```

Comment:
Are these all in a block at the top so we know what we have?

Reply:
Possibly. I was split having a block here or have them with the type. I can change it to your preference.

Reply:
Either way is fine.

Reply:
Then I'll leave it here, less work.